lib/audit/catalog_to_moab.rb

```diff
@@ -81,4 +81,13 @@ def self.check_version_all_dirs_profiled(last_checked_b4_date)
     # update_db_object(preserved_copy) - see #478
   end
 
+  private_class_method def self.moab_found_on_disk?(moab, results)
```

Comment:
I find myself slightly uncomfortable because this isn't a single purpose method and the name doesn't match what it's doing.   
- return true or false depending if moab is on disk  (`online_moab_found`) might be better as `online` is the language we use for the db to distinguish vs. archived??
- add error and report results if moab NOT found

How might we avoid this confusion?

(Note:  I am thinking we should use an INSTANCE of the class to actually do the work of check_catalog_version;  creating the instance inside the loop of `self.check_version_on_dir`)

we could have some methods:

```ruby
def online_moab(druid, storage_dir)
  # code here to instantiate Moab::Storage object from params passed to check_catalog_version
  # e.g.
  moab ||= begin
    object_dir = something_to_manipulate_storage_dir
    moab = Moab::StorageObject.new(druid, object_dir)
  end
end
```

so we then get something like:
```ruby
def check_catalog_version(preserved_copy, storage_dir)
    # TODO: Pohandler.ensure_po_version_matches_this_pc_version (for non-archived, online moab) - see #483

    results = PreservedObjectHandlerResults.new(druid, nil, nil, preserved_copy.endpoint)

    unless online_moab
        results.add_result(PreservedObjectHandlerResults::ONLINE_MOAB_DOES_NOT_EXIST)
        results.report_results
        return
     end
    
      ...

end
```

And further, for switching to an instance, I am thinking something vaguely like

```ruby
  def self.check_version_on_dir(last_checked_b4_date, storage_dir)
    # TODO: ensure last_checked_version_b4_date is in the right format for query - see #485
    pcs = PreservedCopy
          .where('last_version_audit < ? OR last_version_audit IS NULL', last_checked_b4_date)
          .order('last_version_audit IS NOT NULL, last_version_audit ASC')
    pcs.find_each do |pc|
      c2m = CatalogToMoab.new(pc, storage_dir)
      c2m.check_catalog_version
    end
  end
```

Reply:
I'm agreed on having a separation of concerns here.